docs/software/globus-mass-update-procedure.md

```diff
@@ -0,0 +1,185 @@
+
+To get in N-V-R format:
+
+``` console
+[you@host]$ ./osg-outdated-epel-pkgs | \
+    egrep '^(globus|myproxy|gsi)' | \
+    awk 'BEGIN {OFS=""} {print $1, "-", $3}'
+```
+
+or to split up N and V-R in a comma-separated way (which you can feed into a Google Sheet to turn it into two columns):
+
+``` console
+[you@host]$ ./osg-outdated-epel-pkgs | \
+    egrep '^(globus|myproxy|gsi)' | \
+    awk 'BEGIN {OFS=""} {print $1, ",", $3}'
+```
+
+### SVN
+
+Create a separate SVN branch and populate it with all the packages you will update. (Get the list from the doc created above).
+
+``` console
+[you@uw]$ svn mkdir file:///p/vdt/workspace/svn/native/redhat/branches/globus
+### From a checkout, in native/redhat
+[you@uw]$ for x in %RED%<PACKAGES>%ENDCOLOR%; do \
+     svn copy $x branches/globus/${x#trunk/}; \
+   done
+```
+
+### Koji (Mat/Carl)
+
+This requires a Koji administrator. Koji admins as of August 2017 are Mat Selmeci and Carl Edquist.
+
+Ensure Koji tags exist: a destination tag, and a build tag, one for each dver, e.g.:
+
+-   el6-globus
+-   el6-globus-build
+-   el7-globus
+-   el7-globus-build
+
+Set up tag inheritence: base the build tags off of the corresponding `dist-el?-build` tag. This is because we don't want old osg packages interfering with the new versions we're building. These may already exist -- check the `el?-globus-build` tags in the web interface.
+
+``` console
+[you@host]$ for el in el6 el7; do \
+        osg-koji add-tag --parent=dist-$el-build \
+            --arches=x86_64 $el-globus-build; \
+    done
+```
+
+Tag `buildsys-macros` for the OSG release into the build tags:
+
+``` console
+[you@host]$ for el in el6 el7; do \
+       buildsys_macros_nvr=$(osg-koji -q list-tagged osg-3.4-$el-development \
+         buildsys-macros --latest | awk '{print $1}'); \
+       osg-koji tag-pkg $el-globus-build $buildsys_macros_nvr; \
+   done
+```
+
+Ensure Koji targets exist, one for each dver, e.g.:
+
+-   el6-globus (el6-globus-build &rarr; el6-globus)
+-   el7-globus (el7-globus-build &rarr; el7-globus)
+-   kojira-fake-el6-globus (el6-globus &rarr; kojira-fake)
+-   kojira-fake-el7-globus (el7-globus &rarr; kojira-fake)
+
+``` console
+[you@host]$ for el in el6 el7; do \
+       osg-koji add-target $el-globus $el-globus-build $el-globus; \
+       osg-koji add-target kojira-fake-$el-globus $el-globus kojira-fake; \
+   done
+```
+
+If basing the packages off of the Globus repos, add the Globus repos as external repos, and add them to the build tags (but not the dest tags).
+
+Edit `/etc/koji-hub/plugins/sign.conf` and set up the GPG signing for the RPMs. Run `/etc/koji-hub/plugins/fix-permissions` after editing the file.
+
+Per-package work
+----------------
+
+1.  cd into branches/globus
+2.  Download packages from <http://dl.fedoraproject.org/epel/6/SRPMS/>
+
+A useful alias:
+
+``` console
+[you@host]$ alias osg-build-globus="osg-build koji --ktt el6-globus --ktt el7-globus"
+```
```

Comment:
Since the subsequent commands depend on this alias, this should be reworded as a requirement rather than a suggestion for convenience.